### PR TITLE
[Task]: Check usages of Pimcore\Db\Helper::insertOrUpdate and ON DUPLICATE KEY 

### LINKED ISSUE #19: [Task]: Check usages of `Pimcore\Db\Helper::insertOrUpdate` and `ON DUPLICATE KEY`
### Improvement description
Follow up to https://github.com/pimcore/pimcore/issues/13507

Check for usages of `Pimcore\Db\Helper::insertOrUpdate` and `ON DUPLICATE KEY` again, e.g.
https://github.com/pimcore/pimcore/issues/14408

### PR DESCRIPTION
**Pull request by mcop1**
Resolves #19

### CODE REVIEW

File: src/IndexService/Worker/Helper/MySql.php [hunk 1]
```diff
@@ -43,21 +43,37 @@ public function __construct(MysqlConfigInterface $tenantConfig, Connection $db)
...
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getValidTableColumns(string $table, bool $cache = true, bool $primaryKeyColumnsOnly = false): array
     {
         $cacheKey = 'plugin_ecommerce_productindex_columns_' . $table;
 
         if (!Cache\RuntimeCache::isRegistered($cacheKey)) {
```
Comment (kingjia90) [inline src/IndexService/Worker/Helper/MySql.php]: `$cache` isn't in use, was meant to be as following?
```suggestion
        if (!Cache\RuntimeCache::isRegistered($cacheKey) || !$cache) {
```


notes: looks like should be something similar to this structure https://github.com/pimcore/pimcore/blob/230610923ee36ab370261fd13a84619be0162ad8/lib/Model/Dao/AbstractDao.php#L72

File: src/IndexService/Worker/Helper/MySql.php [hunk 2]
```diff
@@ -43,21 +43,37 @@ public function __construct(MysqlConfigInterface $tenantConfig, Connection $db)
...
-                $columns[] = $d['Field'];
+                $fieldName = $d['Field'];
+                $columns[] = $fieldName;
+                if ($d['Key'] === 'PRI') {
+                    $primaryKeyColumns[] = $fieldName;
+                }
             }
-
+            $allColumns = ['columns' => $columns,  'primaryKeyColumns' => $primaryKeyColumns];
             Cache\RuntimeCache::save($columns, $cacheKey);
```
Comment (kingjia90) [inline src/IndexService/Worker/Helper/MySql.php]:
```suggestion
            Cache\RuntimeCache::save($allColumns, $cacheKey);
```

**Review by kingjia90 — CHANGES_REQUESTED** [review]

**Review by mattamon — COMMENTED** [review]
Do we need this for the beta release of the bundle?

**Comment by kingjia90** [comment]
> Do we need this for the beta release of the bundle?

i would say so, the demo beta would be not working properly without ecommerce framework

**Review by kingjia90 — APPROVED** [review]

**Pull request merged**